### ISSUE TITLE
Android Studio Unable to open debugger port (localhost:8700): java.net.SocketException "Connection reset"

### ISSUE THREAD

**Issue by guitar9** [JesusFreke/smali#607]
Unable to open debugger port (localhost:8700): java.net.SocketException "Connection reset"
I get this error when i try it with android studio?
Should i paste the whole folder or just the smali folder in the project?

**Comment by JesusFreke** [comment]
I'm afraid this problem has nothing to do with smalidea :). This is related to IDEA's generic java debugging functionality.

Did you open ddms/android monitor and select the process to debug?

**Comment by guitar9** [comment]
no. but i dont know what i did wrong

![image](https://user-images.githubusercontent.com/25721108/40206283-9d65598a-5a2f-11e8-9d4b-d203d8e05bac.png)

**Comment by JesusFreke** [comment]
In order to debug a process, either the device must be debuggable (ro.debuggable=true), or the app itself must be debuggable (<application ... android:debuggable="true"> in the application's manifest)

**Comment by JesusFreke** [comment]
Also, If you don't have a debuggable device handy, you can use the Android emulator instead.

**Comment by guitar9** [comment]
i have a device. I think i have done something at the begining. When i just import the folder without creating a new project, i have no gradle. So i am not able to run the application

**Comment by JesusFreke** [comment]
You don't need to be able to run the application. In fact, I don't know of a way to set things up that allows you to run it from android studio, because the smalidea plugin doesn't current supporting building/running applications. You have to start the process on the device yourself, and then use ddms to select that process, and then attach android studio's debugger to it

**Comment by guitar9** [comment]
yes but to start the process i have to run or debug the app? And this is what it dont undestand

**Comment by JesusFreke** [comment]
Just run the app. open the app drawer and tap on the app icon, or whatever :)

**Comment by guitar9** [comment]
ok if i run the application i dont see the processess. Somebody know why?
I added  android:debuggable="true" to manifest

**Comment by guitar9** [comment]
ah that was the problem. I changed it in the folder which i imported.

So right way was 

decompile with apktool
build with apktool 
sign the apk 

Now i see the process but how can i debug now? When i place breakpoint it does not stop

**Comment by aqoong** [comment]
I solved it.
But my solution is simple, I do not know why yet.

computer reboot....

Probably because the port connected to the device does not close properly.

**Comment by maiermic** [comment]
I had to change the debug port to `8600`, but you can `Run -> Attach Debugger to Android Process` instead of using a "Remote debug configuration".

**Comment by tanliner** [comment]
> I had to change the debug port to `8600`, but you can `Run -> Attach Debugger to Android Process` instead of using a "Remote debug configuration".

It works well for me.
